### PR TITLE
Simplify callback API

### PR DESCRIPTION
**Pull request by damln**

### CODE REVIEW

File: test/unit/test_callbacks.rb [hunk 1]
```diff
@@ -4,77 +4,98 @@
...
+      Callbacks.register(
+        {"a.b" => [->(name, mutable) { mutable << "Normal #{name}" }]}
+      )
+
+      Callbacks.in('a.b').expose('newOne', ary)
+
+      ary.must_equal ['Normal newOne']
     end
 
     it 'can call a callback in a thread' do
```
[the remark this answers is not in the document]
Comment (damln) [reply test/unit/test_callbacks.rb]: Actually it should stay and we should remove the before block.

File: test/unit/test_callbacks.rb [hunk 2]
```diff
@@ -4,77 +4,98 @@
 module LolSoap
   describe Callbacks do
     before do
-      @lol_callbacks = Callbacks.new.tap do |lc|
-        lc.for('a.b') << ->(name, mutable) { mutable << "Lol #{name}" }
-      end
+      Callbacks.register(
+        {"a.b" => [->(name, mutable) { mutable << "Before #{name}" }]}
+      )
```
[the remark this answers is not in the document]
Comment (damln) [reply test/unit/test_callbacks.rb]: Removed. The before block is useless.

File: lib/lolsoap/callbacks.rb [hunk 3]
```diff
@@ -19,49 +19,33 @@
 #
 class LolSoap::Callbacks
```
[the remark this answers is not in the document]
Comment (damln) [reply lib/lolsoap/callbacks.rb]: Updated.

File: test/unit/test_callbacks.rb [hunk 4]
```diff
@@ -3,78 +3,77 @@
...
+      ary.must_equal ['Route1 TnewOne']
+      yra.must_equal ['Route2 TnewTwo']
     end
 
-    it 'can be enabled' do
-      @lol_callbacks.disable
-      @lol_callbacks.enable
-      temp = Callbacks.new
-      temp.for('a.b') << ->(name, mutable) { mutable << "any #{name} ?" }
+    it 'can be flush' do
```
Comment (danielmbarlow) [inline test/unit/test_callbacks.rb]: Minor detail: `can be flushed`

File: test/unit/test_callbacks.rb [hunk 5]
```diff
@@ -3,78 +3,77 @@
...
 
-    it 'can call a callback' do
-      ary = []
-      Callbacks.in('a.b').expose('lol', ary)
-      ary.must_equal ['Lol lol']
+      Callbacks.store.size.must_equal 1
     end
 
-    it 'can call a callback in a thread' do
+    it 'can call a callback normal' do
```
Comment (danielmbarlow) [inline test/unit/test_callbacks.rb]: Minor detail: How about `can call a callback in the current thread`?

**Comment by danielmbarlow** [comment]
Looks good @damln . I had a look over this because I was wondering how you went about fixing the issue and left a couple of comments on minor details that will make the tests easier to understand. :+1:

File: test/unit/test_callbacks.rb (hunk 4 above)
Comment (damln) [reply test/unit/test_callbacks.rb]: Fixed.

File: test/unit/test_callbacks.rb (hunk 5 above)
Comment (damln) [reply test/unit/test_callbacks.rb]: Fixed.

**Pull request merged**